ext/dom/tests/gh9142.phpt

```diff
@@ -0,0 +1,20 @@
+--TEST--
+GH-9142 (DOMChildNode replaceWith() double-free error when replacing elements not separated by any whitespace)
```

Comment:
I'm confused as to how ``replaceWith`` is getting affected, as it seems only the``before`` and ``after`` methods seem to have fixes. Is it because the replace method "call" the two others ones somehow?

Reply:
replaceWith calls after, and then remove. https://github.com/php/php-src/blob/adb3d5240b2015c1e35543d94b79635e7f5297b7/ext/dom/element.c#L1254-L1255

So spec compliance problem caused problems with replaceWith as well, indirectly.

Reply:
Ahhhhhhh right, that makes snese then :)